src/libxively/tls/xi_tls_layer.c

```diff
@@ -275,18 +280,6 @@ static xi_state_t recv_handler( void* context, void* data, xi_state_t in_out_sta
         return XI_STATE_OK;
     }
 
-    /* if we suppose to reuse the buffer */
-    if ( XI_STATE_WANT_READ == in_out_state && NULL != data_desc )
-    {
```

Comment:
Simply - the layer_data->raw_buffer is tested for existance -  I'm checking if the layer_data->raw_buffer exists != NULL - this is sufficient because the tls_callback_read function copies data from the raw_buffer and deletes it when it's emptied - so if there is a single element on that list it means there is data to read. Does it answer your questions ?